### PR TITLE
WIP: Remove c++98 support, Qt4 and VTK7. Last commit also removes Qt5.5 support.

### PR DESCRIPTION
**Pull request by phcerdan**
COMP: Remove C++98 support

Remove logic involving checks for `CMAKE_CXX_STANDARD=98`

COMP: Remove support for Qt4 and VTK7

Slicer_REQUIRED_QT_VERSION is at least 5.0.0

This allows to remove a lot of logic in form of `#ifdef` involving checking
for QT_VERSION lesser than 5.0.0

COMP: Remove C++98 support

Remove logic involving checks for `CMAKE_CXX_STANDARD=98`

This branch integrates work done in #1060 but with new CMake

Co-authored-by: Hans Johnson <hans-johnson@uiowa.edu>
Co-authored-by: Pablo Hernandez-Cerdan <pablo.hernandez.cerdan@outlook.com>

### CODE REVIEW

**Comment by phcerdan** [comment]
Depends on #1096, so better to compare against that branch for clarity :)

File: Modules/Scripted/DICOMLib/DICOMWidgets.py [hunk 1]
```diff
@@ -1141,10 +1132,10 @@ def configure(self):
     self.setColumnCount(3)
     self.setHorizontalHeaderLabels(['DICOM Data', 'Reader', 'Warnings'])
     self.setSelectionBehavior(qt.QTableView.SelectRows)
-    _setSectionResizeMode(self.horizontalHeader(), qt.QHeaderView.Stretch)
-    _setSectionResizeMode(self.horizontalHeader(), 0, qt.QHeaderView.ResizeToContents)
-    _setSectionResizeMode(self.horizontalHeader(), 1, qt.QHeaderView.Stretch)
-    _setSectionResizeMode(self.horizontalHeader(), 2, qt.QHeaderView.Stretch)
+    self.horizontalHeader(qt.QHeaderView.Stretch)
```
Comment (jcfr) [inline Modules/Scripted/DICOMLib/DICOMWidgets.py]: While crafting the final PR, I noticed that this should be instead be:

```python
self.horizontalHeader().setSectionResizeMode(qt.QHeaderView.Stretch)
self.horizontalHeader().setSectionResizeMode(0, qt.QHeaderView.ResizeToContents)
self.horizontalHeader().setSectionResizeMode(1, qt.QHeaderView.Stretch)
self.horizontalHeader().setSectionResizeMode(2, qt.QHeaderView.Stretch)
```

Comment (phcerdan) [reply Modules/Scripted/DICOMLib/DICOMWidgets.py]: Ups, yes!

Comment (phcerdan) [reply Modules/Scripted/DICOMLib/DICOMWidgets.py]: Changing the wiki...

File: Libs/MRML/Widgets/CMakeLists.txt [hunk 2]
```diff
@@ -396,9 +393,7 @@ if(Slicer_USE_QtTesting)
 endif()
 
   set(_moc_options OPTIONS -DMRML_WIDGETS_HAVE_QT5)
-  if(MRML_WIDGETS_HAVE_WEBKIT_SUPPORT)
```
Comment (jcfr) [inline Libs/MRML/Widgets/CMakeLists.txt]: It seems the next line should be removed

**Comment by jcfr** [comment]
Closing. This is superseded by #1101

**Pull request closed without merging**